src/app/components/Page.js

```diff
@@ -0,0 +1,50 @@
+import React, { Children } from 'react';
+import PropTypes from 'prop-types';
+import { ObserverSections, SubSidebar, usePermissions } from 'react-components';
+import { hasPermission } from 'proton-shared/lib/helpers/permissions';
+
+import Main from './Main';
+import Title from './Title';
+
+const UpgradeNow = () => {
+    return <>Upgrade now !</>;
+};
+
+const Page = ({ config, children }) => {
+    const userPermissions = usePermissions();
+    const { sections = [], permissions: pagePermissions, text } = config;
+
+    if (!hasPermission(userPermissions, pagePermissions)) {
+        return (
+            <Main>
+                <Title>{text}</Title>
+                <UpgradeNow />
+            </Main>
+        );
+    }
+
+    return (
+        <>
+            {sections.length ? <SubSidebar list={sections} /> : null}
+            <Main>
+                <Title>{text}</Title>
+                <ObserverSections>
+                    {Children.map(children, (child, index) => {
+                        const { id = 'related-settings', permissions: sectionPermissions = [] } = sections[index] || {};
```

Comment:
I think defaulting to `related-settings` can make it more confusing than getting undefined. Prefer to define it in the config instead?